### ISSUE TITLE
Graphback inheritance support

### ISSUE THREAD

**Issue by mgobins** [aerogear/graphback#218]
Hello

@wtrocki asked to create issue.
I'm making game. I have 18 types and more more in future.  Graphback works, so I have some problems: 

1. extend (inheritance) do not work nice. Any example with interface? I stopped using interface and it works nice.

2. Permissions are hard to add. I want update to stop for field a, b etc. Code will be removed after generate. @wtrocki said to use custom methods. I created custom methods. It works but it requires additional work. 

3. Bug: Removing custom method leave it in code.
Every time need to remove it in 2 places.

My idea: wrap graphback resolvers. do before after method for permissions. I wrap update methods and do permissions inside.

Really like graphback after implementing server alone, but it needs more docs on auth permissions and custom code.

**Comment by mgobins** [comment]
Sorry for my english. I'm russian.

**Comment by wtrocki** [comment]
Thanks for logging issue. 
TL; DR we need couple issues from this:

Inheritance suppport - not sure what is failing when using interfaces but probably something similar as with enums.

Custom methods needs to be removed from index but still be left just in case - need.to check how this works.

As for authwe have it planned thru graphql modules support but we might kickstart auth specific issue as well

**Comment by wtrocki** [comment]
Another topic I think will be generation that generates code but leaves it in custom. This will mean that graphback will just generate new stuff and always allow people to edit their files. This comes with some drawbacks so we might have separate discussion about it

**Comment by wtrocki** [comment]
@mgobins We have applied all your feedback to separate issues. Thank you so much for contribution!
